app/code/Magento/InventoryCatalog/Model/GetProductIdsBySkus.php

```diff
@@ -44,6 +68,8 @@ public function execute(array $skus): array
             );
         }
 
+        $this->cache->save($this->serializer->serialize($idsBySkus), $cacheKey);
```

Comment:
why do we need serialize/unserialize cache data all the time if our cache is valid only for one HTTP request? 
we don't re-use it for further HTTP requests that's why we no need to persist it